workflows/prognostic_c48_run/runtime/radiation_wrapper.py

```diff
@@ -0,0 +1,484 @@
+        rad_config["lwhtr"] = physics_namelist["lwhtr"]
+        rad_config["lssav"] = default_rad_config["lssav"]
+
+        return rad_config
+
+    def rad_update(
+        self,
+        time: cftime.DatetimeJulian,
+        dt_atmos: float,
+        forcing_dir: str = "./data/forcing",
+    ) -> None:
+        """Update the radiation driver's time-varying parameters"""
+        if self._rad_config is None:
+            raise ValueError(
+                "Radiation driver not initialized. `.rad_init` must be called "
+                "before `.rad_update`."
+            )
+        # idat is supposed to be model initalization time but is unused w/ current flags
+        idat = np.array(
+            [time.year, time.month, time.day, 0, time.hour, time.minute, time.second, 0]
+        )
+        jdat = np.array(
+            [time.year, time.month, time.day, 0, time.hour, time.minute, time.second, 0]
+        )
+        fhswr = np.array(float(self._rad_config["fhswr"]))
+        dt_atmos = np.array(float(dt_atmos))
+        aerosol_data = getdata.aerosol(forcing_dir)
+        _, solar_data = getdata.astronomy(forcing_dir, self._rad_config["isolar"])
+        gas_data = getdata.gases(forcing_dir, self._rad_config["ictmflg"])
+        self._driver.radupdate(
+            idat,
+            jdat,
+            fhswr,
+            dt_atmos,
+            self._rad_config["lsswr"],
+            aerosol_data["kprfg"],
+            aerosol_data["idxcg"],
+            aerosol_data["cmixg"],
+            aerosol_data["denng"],
+            aerosol_data["cline"],
+            solar_data,
+            gas_data,
+        )
+
+    def rad_compute(
+        self,
+        state: State,
+        tracer_metadata: Mapping[str, Any],
+        time: cftime.DatetimeJulian,
+        dt_atmos: float,
+        rank: int,
+        lookup_dir: str = "./data/lookup",
+    ):
+        if self._rad_config is None:
+            raise ValueError(
+                "Radiation driver not initialized. `.rad_init` must be called "
+                "before `.rad_compute`."
+            )
+        if self._input_model is not None:
+            predictions = predict(self._input_model, state)
+            state = UpdatedState(state, predictions)
+        statein = _statein(state, tracer_metadata, self._rad_config["ivflip"])
+        grid, coords = _grid(state)
+        sfcprop = _sfcprop(state)
+        ncolumns, nz = statein["tgrs"].shape[0], statein["tgrs"].shape[1]
+        model = _model(self._rad_config, tracer_metadata, time, dt_atmos, nz, rank)
+        random_numbers = getdata.random(ncolumns, nz, NGPTSW, NGPTLW)
+        lw_lookup = getdata.lw(lookup_dir)
+        sw_lookup = getdata.sw(lookup_dir)
+        out = self._driver._GFS_radiation_driver(
+            model, statein, sfcprop, grid, random_numbers, lw_lookup, sw_lookup
+        )
+        return _unstack(_rename_out(out), coords)
+
+
+class UpdatedState(DerivedFV3State):
```

Comment:
This is just a lazily merged dict object, but the name makes me a bit scared since it implies some other type of state is being added to the fv3net prognostic run. Do you need this object? If so, does it need to be a "DerivedFV3State" or can it be a read-only `Mapping[str, xr.DataArray]`?

Reply:
I'm tempted to take the capability of running the port with ML inputs out of this PR since it was a stub anyway and we're not emphasizing that route right now. If so will delete this.

Reply:
Ended up keeping this capability and adopting your suggestion.

Reply:
I'm a bit nervous about the doubling down on this class. Let's VC about why it is needed?

Reply:
We pair programmed a path to delete this by implementing a `radiation.Radiation.input_variables` property and using it like this:
```


required_names = self._radiation.input_variables
inputs = {name: state[name] for name in required_names}
return {**inputs, **state_updates}
```